src/Saritasa.Tools.SourceGenerator/Models/Metadata/SymbolMetadata.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Text;
+using Saritasa.Tools.SourceGenerator.Abstractions.Models;
+using Saritasa.Tools.SourceGenerator.Infrastructure;
+using Saritasa.Tools.SourceGenerator.Models.Metadata;
+
+namespace Saritasa.Tools.SourceGenerator.Models.Nodes;
+
+/// <summary>
+/// Symbol metadata.
+/// </summary>
+public class SymbolMetadata : SyntaxMetadata
+{
+    /// <summary>
+    /// Symbol name.
+    /// </summary>
+    public string Name { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Symbol namespace.
+    /// </summary>
+    public string Namespace { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Symbol modifier.
+    /// </summary>
+    public string Modifier { get; set; } = MemberModifiers.Public;
+
+    /// <summary>
+    /// Symbol methods.
+    /// </summary>
+    public IReadOnlyCollection<MethodMetadata> Methods { get; set; } = new List<MethodMetadata>();
+
+    /// <summary>
+    /// Symbol members.
+    /// </summary>
+    public IReadOnlyCollection<MemberMetadata> Members { get; set; } = new List<MemberMetadata>();
+
+    /// <summary>
+    /// Interfaces.
+    /// </summary>
+    public IList<string> Interfaces { get; set; } = new List<string>();
+
+    /// <inheritdoc/>
+    public override string Build(IndentWriter writer)
+    {
+        writer.Append("using System.ComponentModel;");
+        writer.AppendLine();
+
+        writer.AppendLine().Append($"namespace {Namespace}");
+        writer.AppendLine().Append("{");
+
+        writer.SetIndent(indentLevel: 1);
```

Comment:
We could also modify the syntax to something like `using (writer.IncreaseIndent()) { ... }` so that we increase the indentation by 1 and then reduce it by 1 after the value is being disposed? This way you won't have to remember to reduce the indentation by 1